docs/plugin/interfaces/guidelines.md

```diff
@@ -0,0 +1,383 @@
+
+!!! note "TODO"
+    Do we want to promote the use of properties in JSONRPC, requirements for the interface specifications.
+
+ Mandatory keywords in an interface definition {#MandatoryKeywordsInAnInterfaceDefinition}
+==============================================
+
+The following keywords are mandatory in a COMRPC interface definition, which is an interface that can be used over process boundaries. The interface definition complies with the C/C++ standard rules for header file definitions and thus these interfaces can (and will) be used directly in C/C++ code.
+
+``` c++
+struct EXTERNAL IObject : public virtual Core::IUnknown {
+    enum { ID = ID_OBJECT };
+};
+```
+
+A ***struct*** in the ***C programming language*** (and many derivatives) is a ***composite data type*** (or ***record***) declaration that defines a physically grouped list of variables under one name in a block of memory, allowing the different variables to be accessed via a single ***pointer*** or by the struct declared name which returns the same address. The struct data type can contain other data types so is used for mixed-data-type records such as a hard-drive directory entry (file length, name, extension, physical address, etc.), or other mixed-type records (name, address, telephone, balance, etc.).
+
+The ```EXTERNAL``` keyword is used for different platforms to define the link type to be used when trying to access this interface. This typically refers to the default constructor/destructors if generated for a struct.
+
+The ```Core::IUnknown``` must always be the base of the interface to allow for interface navigation (```QueryInterface```) and life-time management of the interface (reference counting). As typical implementations of an interface might implement multiple interface by one implementation it is mandatory to all route back to a single lifetime management/Navigational base class (diamond structure) of the core::IUnknown implementation. This is realized by making sure that all interface inherit ```virtual``` from the ```Core::IUnknown```. 
+
+As small-as-it-gets is what Thunder stands for, so identifying an interface by a name (or using a GUID, as Microsoft did) is a bit of overkill. Thunder uses an ID which should be unique. The ID to be assigned to the interface is mandatory and it is defines as ```enum { ID = <NUMBER> };``` the number should be unique and thus all ID’s should be registered in ```ids.h``` to guarantee uniqueness.
+
+The ID’s are split up in ranges. 
+
+|begin|end|Owner|ID storage|
+|--|-|-|-|
+|0x00000000|0x00000080|Thunder internal|```Thunder/Source/plugin/ids.h```|
+|0x00000080|0x80000000|Public Plugin interfaces|```ThunderInterfaces/interfaces/ids.h```|
+|0x80000000|0xA0000000|Custom Plugin interfaces|```<Custom>```|
+|0xA0000000|0xFFFFFFFF|QA Interfaces|```ThunderInterfaces/interfaces/ids.h```|
+
+Tags/annotations available for optimizing the generated code {#TagsAnnotationsAvailableForOptimizingTheGeneratedCode}
```

Comment:
Can we make @text:keep as default for JSON generation. If not can we pls recommend to use @text:keep

Reply:
Hi @anand-ky we cannot make it the default as that would break backwards compatibility for a lot of existing interfaces. 
We think the mandatory rules for these kind of items (e.g. what is mandatory to document, casing used, naming etc.) should be in a separate document that list them for RDK-E, let's discuss this in the next interfaces meeting.